lib/perl/Genome/Config/AnalysisProject/SubjectMapping/Command/Predict/SomaticValidation.pm

```diff
@@ -0,0 +1,162 @@
+        },
+        tumor_sample_common_names => {
+            is      => 'Text',
+            default => Genome::Model::ClinSeq::Command::UpdateAnalysis->__meta__->property('tumor_sample_common_names')->default_value,
+            doc => Genome::Model::ClinSeq::Command::UpdateAnalysis->__meta__->property('tumor_sample_common_names')->doc,
+        },
+        sample_identifier => {
+            is => 'Text',
+            doc => 'The method used to resolve the identifier of the sample.',
+            default => 'name',
+            valid_values => ['name','id'],
+        },
+    ],
+};
+
+sub help_brief {
+    return 'Predict the sample pairings needed to create Somatic Validation models for an AnalysisProject';
+}
+
+sub help_synopsis {
+    return "genome config analysis-project subject-mapping predict somatic-validation <analysis_project_id> <TSV file path>";
+}
+
+sub help_detail {
+    return <<"EOS"
+This command will predict subject mapping for Somatic Validation model types across all samples in an AnalysisProject.
+
+The output file is a 6 column, tab separated format with the following columns:
+
+tumor_subject normal_subject snv_variant_list_id indel_variant_list_id sv_variant_list_id tag
+
+EOS
+}
+
+sub valid_statuses {
+    return ("Pending", "Hold", "In Progress");
+}
+
+my @headers = ('tumor_sample', 'normal_sample','snv_variant_list_id', 'indel_variant_list_id', 'sv_variant_list_id','tag');
+
+sub execute {
+    my $self = shift;
+
+    my $analysis_project = $self->analysis_project;
+
+    my $normal_sample_common_names = $self->normal_sample_common_names;
+    my $tumor_sample_common_names  = $self->tumor_sample_common_names;
+
+    my $id_method = $self->sample_identifier;
+
+    my @all_dna_samples = grep {$_->sample_type =~ /dna/i} $analysis_project->samples;
+    my @dna_samples = List::MoreUtils::uniq(@all_dna_samples);
+    $self->status_message('Found '. scalar(@dna_samples) .' DNA samples');
+
+    my %dna_samples_by_individual_id;
+    for my $dna_sample (@dna_samples) {
+        push @{$dna_samples_by_individual_id{$dna_sample->individual->id}}, $dna_sample;
+    }
+    my @individual_ids = sort keys %dna_samples_by_individual_id;
+    $self->status_message('Found '. scalar(@individual_ids) .' indivudals');
+
+    my @subject_mappings;
+    for my $individual_id (@individual_ids) {
+        my $individual = Genome::Individual->get($individual_id);
+        my @individual_dna_samples = @{$dna_samples_by_individual_id{$individual_id}};
+        my $individual_dna_sample_count = scalar(@individual_dna_samples);
+        $self->status_message('Found '. $individual_dna_sample_count .' samples for individual '. $individual->__display_name__);
+
+        if ( $individual_dna_sample_count > 2) {
+            $self->fatal_message('This command currently does not support more than 2 DNA samples per individual.');
+        }
```

Comment:
Should we warn about this individual but continue to create the rest of the predictions?